tools/autograd/derivatives.yaml

```diff
@@ -640,15 +640,15 @@
   self: mvlgamma_backward(grad, self, p)
 
 - name: native_batch_norm(Tensor input, Tensor? weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, bool training, float momentum, float eps) -> (Tensor, Tensor, Tensor)
-  input, weight, bias: native_batch_norm_backward(grad, input, weight, running_mean, running_var, result1, result2, training, eps, grad_input_mask)
+  input, weight, bias: "grad.defined() ? native_batch_norm_backward(grad, input, weight, running_mean, running_var, result1, result2, training, eps, grad_input_mask) : std::tuple<Tensor, Tensor, Tensor>()"
```

Comment:
This function has multiple outputs. Is only one differentiable? Or should you be using `grads[i]` here?

Reply:
It calculates multiple input grads, but only uses one output grad for that calculation. Notice that `native_batch_norm_backward` only has `grad` as one of its arguments, not `grads[i]`.

Reply:
Ok, so only the first output is differentiable and the others are non-differentiable. Sounds good.